### PR TITLE
Support Lro interim response

### PR DESCRIPTION
**Pull request by Yao725**
# Description
<i>Fix https://github.com/Azure/azure-sdk-for-net/issues/29060
Regen PR https://github.com/Azure/azure-sdk-for-net/pull/29638</i>

# Checklist

To ensure a quick review and merge, please ensure:
- [ ] The PR has a understandable title and description explaining the _why_ and _what_.
- [ ] The PR is opened in draft if not ready for review yet.
   - If opened in draft, please allocate sufficient time (24 hours) after moving out of draft for review
- [ ] The branch is recent enough to not have merge conflicts upon creation.

# Ready to Land?
- [ ] Build is completely green
   - Submissions with test failures require tracking issue and approval of a CODEOWNER
- [ ] At least one +1 review by a CODEOWNER
- [ ] All -1 reviews are confirmed resolved by the reviewer 
   - Override/Marking reviews stale must be discussed with CODEOWNERS first

### CODE REVIEW

File: samples/Azure.Management.Storage/Generated/LongRunningOperation/BlobRestoreStatusRestoreBlobRangesOperation.cs [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+using System.Threading.Tasks;
+using Azure;
+using Azure.Core;
+using Azure.Core.Pipeline;
+using Azure.Management.Storage.Models;
+using Azure.ResourceManager;
+
+namespace Azure.Management.Storage
+{
+    public class BlobRestoreStatusRestoreBlobRangesOperation : ArmOperation<BlobRestoreStatus>
```
Comment (m-nash) [inline samples/Azure.Management.Storage/Generated/LongRunningOperation/BlobRestoreStatusRestoreBlobRangesOperation.cs:19]: name should be [Resource][Method]Operation

File: samples/Azure.Management.Storage/readme.md [hunk 2]
```diff
@@ -43,6 +43,9 @@ rename-mapping:
   BlobServiceProperties: BlobService
   FileServiceProperties: FileService
 
+enable-lro-interim-status:
```
Comment (m-nash) [inline samples/Azure.Management.Storage/readme.md:46]: This also works with the swagger setting right?  If so can we simply do that directive here instead and remove this configuration as eventually we want this to bubble up into the swagger / cadl and not live in customization.

**Review by m-nash — APPROVED** [review]
approve after comments are addressed

File: samples/Azure.Management.Storage/Generated/LongRunningOperation/BlobRestoreStatusRestoreBlobRangesOperation.cs (hunk 1 above)
Comment (Yao725) [reply samples/Azure.Management.Storage/Generated/LongRunningOperation/BlobRestoreStatusRestoreBlobRangesOperation.cs]: fixed

File: samples/Azure.Management.Storage/readme.md (hunk 2 above)
Comment (Yao725) [reply samples/Azure.Management.Storage/readme.md]: I changed the implementation to use the [swagger extension setting](https://github.com/Azure/autorest.csharp/blob/9a3032f7bc84eeb8656e97da0c7ae7a53c9c05cb/src/AutoRest.CSharp/Common/Input/CodeModelPartials.cs#L43).

**Pull request merged**